### DISCUSSION TITLE
could not select device driver "nvidia" with capabilities: [[gpu]]

### DISCUSSION THREAD

**Discussion by alansenairj** [open-webui/open-webui#1319] [Q&A]
This is the first time I am running open-webui from a docker compose stack with GPU supports. When it is installing, I got this message and I got any container running

[+] Running 0/5
 ⠼ Network open-webui_default      Created                                                                          0.4s 
 ⠼ Volume "open-webui_ollama"      Created                                                                          0.4s 
 ⠼ Volume "open-webui_open-webui"  Created                                                                          0.4s 
 ⠼ Container ollama                Starting                                                                         0.4s 
 ⠸ Container open-webui            Created                                                                          0.3s 
Error response from daemon: could not select device driver "nvidia" with capabilities: [[gpu]]

I am using Fedora 39 with RTX2060 super using Nvidia driver. 

 nvidia-installer -v |grep version                        
nvidia-installer:  version 545.29.06

nvidia-smi                                                  
Tue Mar 26 22:58:15 2024       
+---------------------------------------------------------------------------------------+
| NVIDIA-SMI 545.29.06              Driver Version: 545.29.06    CUDA Version: 12.3     |
|-----------------------------------------+----------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |         Memory-Usage | GPU-Util  Compute M. |
|                                         |                      |               MIG M. |
|=========================================+======================+======================|
|   0  NVIDIA GeForce RTX 2060 ...    Off | 00000000:01:00.0  On |                  N/A |
| 34%   52C    P0              35W / 175W |   1726MiB /  8192MiB |     10%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+
                                                                                         
+---------------------------------------------------------------------------------------+
| Processes:                                                                            |
|  GPU   GI   CI        PID   Type   Process name                            GPU Memory |
|        ID   ID                                                             Usage      |
|=======================================================================================|
|    0   N/A  N/A      2370      G   /usr/bin/kwalletd5                            2MiB |
|    0   N/A  N/A      2372      G   /usr/libexec/Xorg                           855MiB |
|    0   N/A  N/A      2901      G   /usr/bin/ksmserver                            2MiB |
|    0   N/A  N/A      2903      G   /usr/bin/kded5                                2MiB |
|    0   N/A  N/A      2906      G   /usr/bin/kwin_x11                            41MiB |
|    0   N/A  N/A      2945      G   /usr/bin/plasmashell                         78MiB |
|    0   N/A  N/A      2985      G   ...5/polkit-kde-authentication-agent-1        2MiB |
|    0   N/A  N/A      2987      G   /usr/libexec/xdg-desktop-portal-kde           2MiB |
|    0   N/A  N/A      3133      G   /usr/libexec/kdeconnectd                      2MiB |
|    0   N/A  N/A      3144      G   /usr/bin/keepassxc                            2MiB |
|    0   N/A  N/A      3184      G   /usr/bin/kaccess                              2MiB |
|    0   N/A  N/A      3194      G   /usr/libexec/DiscoverNotifier                 2MiB |
|    0   N/A  N/A      3196      G   /usr/bin/kalendarac                           2MiB |
|    0   N/A  N/A      3449      G   /usr/bin/python3                              2MiB |
|    0   N/A  N/A      3530      G   /usr/bin/akonadi_control                      2MiB |
|    0   N/A  N/A      3563      G   ...ures=SpareRendererForSitePerProcess        2MiB |
|    0   N/A  N/A      3757      G   /usr/bin/akonadi_akonotes_resource            2MiB |
|    0   N/A  N/A      3758      G   /usr/bin/akonadi_archivemail_agent            2MiB |
|    0   N/A  N/A      3759      G   /usr/bin/akonadi_birthdays_resource           2MiB |
|    0   N/A  N/A      3760      G   /usr/bin/akonadi_contacts_resource            2MiB |
|    0   N/A  N/A      3761      G   .../bin/akonadi_followupreminder_agent        2MiB |
|    0   N/A  N/A      3762      G   /usr/bin/akonadi_ical_resource                2MiB |
|    0   N/A  N/A      3763      G   /usr/bin/akonadi_indexing_agent               2MiB |
|    0   N/A  N/A      3764      G   /usr/bin/akonadi_maildir_resource             2MiB |
|    0   N/A  N/A      3765      G   /usr/bin/akonadi_maildispatcher_agent         2MiB |
|    0   N/A  N/A      3766      G   /usr/bin/akonadi_mailfilter_agent             2MiB |
|    0   N/A  N/A      3767      G   /usr/bin/akonadi_mailmerge_agent              2MiB |
|    0   N/A  N/A      3769      G   /usr/bin/akonadi_migration_agent              2MiB |
|    0   N/A  N/A      3773      G   /usr/bin/akonadi_newmailnotifier_agent        2MiB |
|    0   N/A  N/A      3774      G   /usr/bin/akonadi_sendlater_agent              2MiB |
|    0   N/A  N/A      3775      G   /usr/bin/akonadi_unifiedmailbox_agent         2MiB |
|    0   N/A  N/A      3975      G   /opt/google/chrome/chrome                     2MiB |
|    0   N/A  N/A      4288      G   ...bin/plasma-browser-integration-host        2MiB |
|    0   N/A  N/A      4709      G   /usr/lib/virtualbox/VirtualBox                2MiB |
|    0   N/A  N/A      8170      G   /usr/libexec/baloorunner                      2MiB |
|    0   N/A  N/A      8216      G   ...rdCache,DocumentPictureInPictureAPI      104MiB |
|    0   N/A  N/A     10669      G   ...b64/libreoffice/program/soffice.bin        2MiB |
|    0   N/A  N/A     10940      G   /usr/bin/gpclient                             3MiB |
|    0   N/A  N/A     15587      G   /usr/bin/kwrite                               2MiB |
|    0   N/A  N/A     93722      G   /usr/lib/virtualbox/VirtualBoxVM              2MiB |
|    0   N/A  N/A     97580      G   localsend                                    22MiB |
|    0   N/A  N/A    115965      G   /usr/bin/dolphin                              2MiB |
|    0   N/A  N/A    129044      G   /usr/bin/konsole                              2MiB |
|    0   N/A  N/A    139271      G   ...ures=SpareRendererForSitePerProcess       76MiB |
|    0   N/A  N/A    139274      G   ...sion,SpareRendererForSitePerProcess       61MiB |
|    0   N/A  N/A    139313      G   ...seed-version=20240325-114740.111000      185MiB |
|    0   N/A  N/A    140463      G   ...erProcess --variations-seed-version        2MiB |
|    0   N/A  N/A    191025      G   /usr/lib64/firefox/firefox                  186MiB |
+---------------------------------------------------------------------------------------+

I really don't know where to start. Can someone give me some help to put it to run?

**Comment by Rock-son** [comment]
There is a short mention in How to install that you need Nvidia CUDA container toolkit: https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/latest/install-guide.html#prerequisites (installation and configuration)

**Reply by ImGGAAVVIINN** [reply]
I already have the tool kit installed, still not fixed
